### PR TITLE
Fix PACK_MAP for unsigned types

### PR DESCRIPTION
**Pull request by k0kubun**
I think void* and unsigned types should be packed/unpacked as unsigned values.

### CODE REVIEW

File: lib/fiddle/pack.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ module PackInfo # :nodoc: all
       ALIGN_MAP[TYPE_LONG_LONG] = ALIGN_MAP[-TYPE_LONG_LONG] = ALIGN_LONG_LONG
       PACK_MAP[TYPE_LONG_LONG] = PACK_MAP[-TYPE_LONG_LONG] = "q"
```
Comment (kou) [inline lib/fiddle/pack.rb:49]: Should we use `Q` for `-TYPE_LONG_LONG` too?

Comment (k0kubun) [reply lib/fiddle/pack.rb]: Good catch, thank you https://github.com/ruby/fiddle/pull/110/commits/43b1fe4cfd95b719d8bef67f3b49d60f439cfd9e

**Review by kou — APPROVED** [review]
+1

**Comment by k0kubun** [comment]
FYI, I cherry-picked this to ruby/ruby https://github.com/ruby/ruby/commit/106744107b485a371ed788fe590c6ce778de16df as I needed to use this in ruby/ruby (MJIT).

I'd also like to cherry-pick https://github.com/ruby/fiddle/pull/111 if you're okay with it.

**Comment by kou** [comment]
I'm OK with it.
You can merge the current master instead of cherry-picking some commits if you want.

**Comment by k0kubun** [comment]
got it, thank you!

**Pull request merged**